### PR TITLE
Rename py long  > py int #4260

### PR DESCRIPTION
**Pull request by shreyash2002**
Thank you for contributing to PyO3!

By submitting these contributions you agree for them to be dual-licensed under PyO3's [MIT OR Apache-2.0 license](https://github.com/PyO3/pyo3#license).

Please consider adding the following to your pull request:
 - an entry for this PR in newsfragments - see [https://pyo3.rs/main/contributing.html#documenting-changes]
   - or start the PR title with `docs:` if this is a docs-only change to skip the check
 - docs to all new functions and / or detail in the guide
 - tests for all new or changed functions

PyO3's CI pipeline will check your pull request, thus make sure you have checked the `Contributing.md` guidelines. To run most of its tests
locally, you can run ```nox```. See ```nox --list-sessions```
for a list of supported actions.

### CODE REVIEW

**Comment by shreyash2002** [comment]
@davidhewitt 
I have created a draft PR with some changes. Can you review it and suggest what more i should do?

File: src/types/num.rs [hunk 1]
```diff
@@ -7,6 +7,6 @@ use crate::{ffi, PyAny};
 /// and [`extract`](super::PyAnyMethods::extract)
 /// with the primitive Rust integer types.
 #[repr(transparent)]
-pub struct PyLong(PyAny);
+pub struct PyInt(PyAny);
 
-pyobject_native_type_core!(PyLong, pyobject_native_static_type_object!(ffi::PyLong_Type), #checkfunction=ffi::PyLong_Check);
+pyobject_native_type_core!(PyInt, pyobject_native_static_type_object!(ffi::PyInt_Type), #checkfunction=ffi::PyInt_Check);
```
Comment (LilyFoote) [inline src/types/num.rs:12]: I think it's only `PyLong` that should change, not `ffi::PyLong_Type` or `ffi::PyLong_Check`:

```suggestion
pyobject_native_type_core!(PyInt, pyobject_native_static_type_object!(ffi::PyLong_Type), #checkfunction=ffi::PyLong_Check);
```

File: pyo3-ffi/README.md [hunk 2]
```diff
@@ -95,15 +95,15 @@ pub unsafe extern "C" fn sum_as_string(
     }
 
     let arg1 = *args;
-    if PyLong_Check(arg1) == 0 {
+    if PyInt_Check(arg1) == 0 {
```
Comment (LilyFoote) [inline pyo3-ffi/README.md:98]:
```suggestion
    if PyLong_Check(arg1) == 0 {
```

File: pyo3-ffi/README.md [hunk 3]
```diff
@@ -95,15 +95,15 @@ pub unsafe extern "C" fn sum_as_string(
...
         PyErr_SetString(
             PyExc_TypeError,
             c_str!("sum_as_string() expected an int for positional argument 1").as_ptr(),
         );
         return std::ptr::null_mut();
     }
 
-    let arg1 = PyLong_AsLong(arg1);
+    let arg1 = PyInt_AsLong(arg1);
```
Comment (LilyFoote) [inline pyo3-ffi/README.md:106]:
```suggestion
    let arg1 = PyLong_AsLong(arg1);
```

File: pyo3-ffi/README.md [hunk 4]
```diff
@@ -117,7 +117,7 @@ pub unsafe extern "C" fn sum_as_string(
         return std::ptr::null_mut();
     }
 
-    let arg2 = PyLong_AsLong(arg2);
+    let arg2 = PyInt_AsLong(arg2);
```
Comment (LilyFoote) [inline pyo3-ffi/README.md:120]:
```suggestion
    let arg2 = PyLong_AsLong(arg2);
```

**Review by LilyFoote — COMMENTED** [review]
Hey, thanks for taking a look at this.

I don't think you should need to change any code in `pyo3-ffi` for this issue. Once you've reverted those changes, I suspect you'll start getting more helpful error messages about which bits of the codebase still do need to change.

**Comment by MatthijsKok** [comment]
Superseded by #4347

**Comment by davidhewitt** [comment]
Thanks all!

**Pull request closed without merging**